### PR TITLE
Example app makeover

### PR DESCRIPTION
**Pull request by doregg**
Support hosted/embedded mode in one value change (isomorphic login/logout buttons & effect & provider configuration)
Pages:

- Home page
- entitlement page
- step up with no max age page (button)
- step up with small max age (button)
- step up with high max age (button)
- step up - full page (must be stepped up without a button)
- Authorized content
- Private route (legacy)
- Fallback route

Risk: There is no option to run admin-box tests against this branch, so after this merge, they can fail and block the version branch.
I plan to merge it and check it immediately in the admin box. Worst case - revert

### CODE REVIEW

File: projects/example/src/app/app.component.ts [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
   templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
 })
 export class AppComponent implements OnInit, OnDestroy {
   isLoading = true;
   loadingSubscription: Subscription;
+  isAuthenticatedSubscription: Subscription;
+  userSubscription?: Subscription;
+  authenticated = false;
+  user?: any;
```
Comment (yuvalotem1) [inline projects/example/src/app/app.component.ts:18]: `import { User } from '@frontegg/redux-store';
`
  user?: User;

File: projects/example/src/app/app.component.ts [hunk 2]
```diff
@@ -1,24 +1,63 @@
...
+    this.router.navigateByUrl(this.fronteggAppService.authRoutes.logoutUrl);
+  }
+
+  loginWithRedirect(): void {
+    this.fronteggAuthService.loginWithRedirect();
   }
 
   ngOnInit(): void {
-    console.log('AppComponent', 'ngOnInit');
+    this.userSubscription = this.fronteggAuthService.user$.subscribe((user: any) => {
```
Comment (yuvalotem1) [inline projects/example/src/app/app.component.ts:53]:
```suggestion
    this.userSubscription = this.fronteggAuthService.user$.subscribe((user: User) => {
```

File: projects/example/src/app/components/entitlements-page/entitlements-page.component.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  entitlementsResults: {
+    [key: string]: { 
+      name: string,
+      isEntitled: Entitlement['isEntitled'],
+      justification?: Entitlement['justification']
+    } 
+  } = {};
+
+  user?: any;
```
Comment (yuvalotem1) [inline projects/example/src/app/components/entitlements-page/entitlements-page.component.ts:28]: `User`

File: projects/example/src/app/components/entitlements-page/entitlements-page.component.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    const entitlementsRequests = [
+      { queryType: EntitlementsQueryType.FEATURE, arg: 'sso', },
+      { queryType: EntitlementsQueryType.FEATURE, arg: 'sso', customAttributes: { env: 'dev' } },
+      { queryType: EntitlementsQueryType.ENTITLEMENTS, arg: { featureKey: 'proteins.*' }, customAttributes: { pro: '20gr' }},
+      { queryType: EntitlementsQueryType.PERMISSION, arg: 'dora.protein.*', },
+      { queryType: EntitlementsQueryType.ENTITLEMENTS, arg: { permissionKey: 'fe.secure.*' } },
+      { queryType: EntitlementsQueryType.PERMISSION, arg: 'fe.secure.*', customAttributes: { env: 'dev' } },
+    ];
+
+    const subscriptionGenerators: any = {
```
Comment (yuvalotem1) [inline projects/example/src/app/components/entitlements-page/entitlements-page.component.ts:54]: maybe remove any

File: projects/example/src/app/components/entitlements-page/entitlements-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    });
+
+    this.feedEntitlements();
+  }
+
+  feedEntitlements() {
+    const entitlementsRequests = [
+      { queryType: EntitlementsQueryType.FEATURE, arg: 'sso', },
+      { queryType: EntitlementsQueryType.FEATURE, arg: 'sso', customAttributes: { env: 'dev' } },
+      { queryType: EntitlementsQueryType.ENTITLEMENTS, arg: { featureKey: 'proteins.*' }, customAttributes: { pro: '20gr' }},
```
Comment (yuvalotem1) [inline projects/example/src/app/components/entitlements-page/entitlements-page.component.ts:48]: what is `20gr` ?

**Review by yuvalotem1 — APPROVED** [review]

File: projects/example/src/app/components/entitlements-page/entitlements-page.component.ts (hunk 4 above)
Comment (doregg) [reply projects/example/src/app/components/entitlements-page/entitlements-page.component.ts:54]: ts is yelling

**Pull request merged**